src/gsModeling/gsPatchGenerator.h

```diff
@@ -124,7 +124,7 @@ void gsPatchGenerator<T>::preparePatch(gsTensorBSplineBasis<d,T> & resultBasis,
 
     std::vector<Boundary_t*> input;// permutation of the input boundaries
     input.reserve(2*d);
-    std::vector<int> perm;// permutation to be computed
+    std::vector<index_t> perm;// permutation to be computed
```

Comment:
short_t or int fits better here.

Reply:
fails at runtime of `gsPatchFromBoundary` if perm is set to `std::vector<short_t>`, have no idea how to fix the assignment to `a`.
https://github.com/gismo/gismo/blob/a5b765bc2809f181ab008863b78146394c7ebdfd/src/gsCore/gsMultiPatch.hpp#L188-L193